packages/protocol/contracts/governance/VestingInstance.sol

```diff
@@ -0,0 +1,379 @@
+  }
+
+  /**
+   * @notice A wrapper func for the lock gold method
+   * @param value the value of gold to be locked
+   * @dev To be called only by the beneficiary of the vesting
+   */
+  function lockGold(uint256 value) external nonReentrant onlyBeneficiary {
+    // the beneficiary may not lock more than the vesting currently has available
+    require(
+      value <= getGoldToken().balanceOf(address(this)),
+      "Gold amount to lock is greater than the currently vested amount"
+    );
+    getLockedGold().lock.gas(gasleft()).value(value)();
+  }
+
+  /**
+   * @notice A wrapper func for the unlock gold method function
+   * @param value the value of gold to be unlocked for the vesting instance
+   * @dev To be called only by the beneficiary of the vesting
+   */
+  function unlockGold(uint256 value) external nonReentrant onlyBeneficiary {
+    getLockedGold().unlock(value);
+  }
+
+  /**
+   * @notice A wrapper func for the relock locked gold method function
+   * @param index the index of the pending locked gold withdrawal
+   * @param value the value of gold to be relocked for the vesting instance
+   * @dev To be called only by the beneficiary of the vesting.
+   */
+  function relockGold(uint256 index, uint256 value) external nonReentrant onlyBeneficiary {
+    getLockedGold().relock(index, value);
+  }
+
+  /**
+   * @notice A wrapper func for the withdraw locked gold method function
+   * @param index the index of the pending locked gold withdrawal
+   * @dev To be called only by the beneficiary of the vesting. The amount shall be withdrawn back to the vesting instance
+   */
+  function withdrawLockedGold(uint256 index) external nonReentrant onlyBeneficiary {
+    getLockedGold().withdraw(index);
+  }
+
+  /**
+   * @notice A wrapper func for the authorize vote signer account method
+   * @param signer The address of the signing key to authorize.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev To be called only by the beneficiary of the vesting. The v,r and s signature should be a signed message by the beneficiary being the authorized address
+   */
+  function authorizeVoteSigner(address signer, uint8 v, bytes32 r, bytes32 s)
+    external
+    nonReentrant
+    onlyBeneficiary
+  {
+    getAccounts().authorizeVoteSigner(signer, v, r, s);
+  }
+
+  /**
+   * @notice A wrapper func for the authorize validator signer account method
+   * @param signer The address of the signing key to authorize.
+   * @param v The recovery id of the incoming ECDSA signature.
+   * @param r Output value r of the ECDSA signature.
+   * @param s Output value s of the ECDSA signature.
+   * @dev To be called only by the beneficiary of the vesting.
+   */
+  function authorizeValidatorSigner(address signer, uint8 v, bytes32 r, bytes32 s)
```

Comment:
So I think we actually want to remove this one, so that users are not able to create validators with gold that is not "theirs" yet, as it could be slashed

Reply:
done